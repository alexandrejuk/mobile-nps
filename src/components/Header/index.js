import React from 'react'
import PropTypes from 'prop-types'
import CompanyLogo from '../../assets/images/logo.svg'
import arrowIcon from '../../assets/icons/arrow-left.svg'
import styles from './style.module.css'

const Header = ({
  action,
}) => (
  <div className={styles.header}>
    <div
      className={styles.icon}
      onClick={action}
      role="button"
    >
      <img src={arrowIcon} alt="action go back" />
    </div>
    <div className={styles.logo}>
      <img src={CompanyLogo} alt="company logo" />
    </div>
  </div>
)

Header.propTypes = {
  action: PropTypes.func.isRequired,
}

export default Header
