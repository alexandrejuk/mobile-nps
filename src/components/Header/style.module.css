@import '../../assets/css/colors.css';

.header {
  align-items: center;
  display:flex;
  background-color: var(--color-white);
  border-bottom: solid 1px var( --color-alto);
  box-sizing: border-box;
  height: 60px;
  justify-content: space-between;
  padding: 5px;
}

.logo {
  margin: auto;
  width: 77px;
}
