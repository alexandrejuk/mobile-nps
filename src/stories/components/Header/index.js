import React from 'react'
import { action } from '@storybook/addon-actions'
import { Header } from '../../../components'

export default {
  title: 'Components/Header',
}

export const Default = () => (
  <Header
    action={action('go back!')}
  />
)
