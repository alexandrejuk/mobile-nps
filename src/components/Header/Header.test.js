import React from 'react'
import { render } from '@testing-library/react'

import Header from './'

describe('Header component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Header
        action={() => {}}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
