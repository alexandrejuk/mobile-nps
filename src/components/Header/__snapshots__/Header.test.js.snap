// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div>
  <div
    class="header"
  >
    <div
      class="icon"
      role="button"
    >
      <img
        alt="action go back"
        src="arrow-left.svg"
      />
    </div>
    <div
      class="logo"
    >
      <img
        alt="company logo"
        src="logo.svg"
      />
    </div>
  </div>
</div>
`;
